src/librustc/ty/util.rs

```diff
@@ -861,11 +861,13 @@ fn is_copy_raw<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
     let (param_env, ty) = query.into_parts();
     let trait_def_id = tcx.require_lang_item(lang_items::CopyTraitLangItem);
     tcx.infer_ctxt()
-       .enter(|infcx| traits::type_known_to_meet_bound(&infcx,
-                                                       param_env,
-                                                       ty,
-                                                       trait_def_id,
-                                                       DUMMY_SP))
+        .enter(|infcx| traits::type_known_to_meet_bound_modulo_regions(
```

Comment:
Should we advertise and/or track this behavior more strongly, or even consider fixing it?

I remember seeing something about MIR typeck/borrowck being able to take into account the fact that `Copy` impls can impose constraints on/between lifetimes.

Maybe we should be conservative and require that no lifetime constraints are needed to satisfy the `Copy` bound, for methods like these?
I guess I don't know what's using them anymore.

Reply:
Urgh, I'm not quite sure what you meant, @eddyb, when you said "fixing it". I guess you mean -- decide whether or not to consider the type to be copy? 

I am not sure if we can do that, nor if we want to. My goal is for the lowering to MIR is something to be done with erased regions, which implies we should be able to decide "copy or not" on that basis. Then we reconstruct regions entirely in the NLL pass (the NLL pass currently uses *nothing* from type check except for which regions are bound -- in my ideal world, we'd infer that *too*, but I've not put serious thought into it).

The NLL type checker however does enforce the full set of constraints about regions when you try to copy a type.

Reply:
Sure, what I mean is that these e.g. `Ty::is_copy` methods are far too general and will get misused if we leave them like this.
So I would prefer if this sort of thing was more opt-in at the call sites, if possible.

Reply:
I will rename them for now to `is_copy_modulo_regions` or something